skfda/preprocessing/dim_reduction/projection/_fpca.py

```diff
@@ -1,33 +1,445 @@
+        n_components (int): number of principal components to obtain from
+            functional principal component analysis. Defaults to 3.
+        centering (bool): if True then calculate the mean of the functional data
+            object and center the data first
+    """
+
+    def __init__(self, n_components=3, centering=True):
+        """FPCA constructor
+
+        Args:
+            n_components (int): number of principal components to obtain from
+                functional principal component analysis
+            centering (bool): if True then calculate the mean of the functional
+                data object and center the data first. Defaults to True
+        """
+        self.n_components = n_components
+        self.centering = centering
+
+    @abstractmethod
+    def fit(self, X, y=None):
+        """Computes the n_components first principal components and saves them
+        inside the FPCA object.
+
+        Args:
+            X (FDataGrid or FDataBasis):
+                the functional data object to be analysed
+            y (None, not used):
+                only present for convention of a fit function
+
+        Returns:
+            self (object)
+        """
+        pass
+
+    @abstractmethod
+    def transform(self, X, y=None):
+        """Computes the n_components first principal components score and
+        returns them.
+
+        Args:
+            X (FDataGrid or FDataBasis):
+                the functional data object to be analysed
+            y (None, not used):
+                only present because of fit function convention
+
+        Returns:
+            (array_like): the scores of the data with reference to the
+            principal components
+        """
+        pass
+
+    def fit_transform(self, X, y=None, **fit_params):
+        """Computes the n_components first principal components and their scores
+        and returns them.
+        Args:
+            X (FDataGrid or FDataBasis):
+                the functional data object to be analysed
+            y (None, not used):
+                only present for convention of a fit function
+
+        Returns:
+            (array_like): the scores of the data with reference to the
+            principal components
+        """
+        self.fit(X, y)
+        return self.transform(X, y)
+
+
+class FPCABasis(FPCA):
+    """Functional principal component analysis for functional data represented
+    in basis form.
+
+    Attributes:
+        components_ (FDataBasis): this contains the principal components in a
+            basis representation.
+        component_values_ (array_like): this contains the values (eigenvalues)
+            associated with the principal components.
+        pca_ (sklearn.decomposition.PCA): object for PCA.
+            In both cases (discretized FPCA and basis FPCA) the problem can be
+            reduced to a regular PCA problem and use the framework provided by
+            sklearn to continue.
+
+    Parameters:
```

Comment:
Reverse the order: first parameters, then attributes.

Reply:
Done